htdocs/societe/class/societe.class.php

```diff
@@ -4082,12 +4082,16 @@ public function isACompany()
 			$isacompany = 1;
 		} elseif (!empty($this->idprof1) || !empty($this->idprof2) || !empty($this->idprof3) || !empty($this->idprof4) || !empty($this->idprof5) || !empty($this->idprof6)) {
 			$isacompany = 1;
-		} elseif (!empty($this->typent_code) && $this->typent_code != 'TE_UNKNOWN') {
-			// TODO Add a field is_a_company into dictionary
-			if (preg_match('/^TE_PRIVATE/', $this->typent_code)) {
-				$isacompany = 0;
+		} else {
+			if (getDolGlobalString('DEFINE_CUSTOMERS_ARE_COMPANIES_BY_TYPE_COMPANY')) {
+				// TODO Add a field is_a_company into dictionary
+				if (preg_match('/^TE_PRIVATE/', $this->typent_code)) {
```

Comment:
Whats if typent_code is not defined (null) it may generzte an error ?
We may add before
If !empty() && ...

Reply:
Passing null as a $subject for preg_match() does not raise an error. It returns 0. I tested it.

```
echo '<?= var_dump(preg_match("/truc/", null)) ?>' > /tmp/test.php ; php /tmp/test.php
```

And $this->typent_code is always set (even if null) since it is a property of class Societe, so php 8.X won't cry either.